sdl_android/src/main/java/com/smartdevicelink/protocol/SdlProtocol.java

```diff
@@ -0,0 +1,1319 @@
+                        // specific service type
+                        for(int transportNum : transportPriorityForServiceMap.get(secondaryService)){
+                            if(transportNum == PRIMARY_TRANSPORT_ID){
+                                break; // Primary is favored for this service type, break out...
+                            }else if(transportNum == SECONDARY_TRANSPORT_ID){
+                                // The secondary transport can be used to start this service
+                                activeTransports.put(secondaryService, transportRecord);
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
+        }else{
+            Log.d(TAG, transportRecord.toString() + " transport was NOT registered!");
+        }
+        //Notify any listeners for this secondary transport
+        List<ISecondaryTransportListener> listenerList = secondaryTransportListeners.remove(transportRecord.getType());
+        if(listenerList != null){
+            for(ISecondaryTransportListener listener : listenerList){
+                if(registered) {
+                    listener.onConnectionSuccess(transportRecord);
+                }else{
+                    listener.onConnectionFailure();
+                }
+            }
+        }
+
+        if(DebugTool.isDebugEnabled()){
+            printActiveTransports();
+        }
+    }
+
+    public void onTransportsConnectedUpdate(List<TransportRecord> transports){
+        Log.d(TAG, "Connected transport update");
+
+        //Temporary: this logic should all be changed to handle multiple transports of the same type
+        ArrayList<TransportType> connectedTransports = new ArrayList<>();
+        for(TransportRecord record: transports){
+            connectedTransports.add(record.getType());
+        }
+
+        if(connectedPrimaryTransport != null && !connectedTransports.contains(connectedPrimaryTransport.getType())){
+            //The primary transport being used is no longer part of the connected transports
+            //The transport manager callbacks should hadle the disconnect code
+            connectedPrimaryTransport = null;
+            return;
+        }
+
+        if(activeTransports.get(SessionType.RPC) == null){
+            //There is no currently active transport for the RPC service meaning no primary transport
+            TransportRecord preferredPrimaryTransport = getPreferredTransport(requestedPrimaryTransports,transports);
+            if(preferredPrimaryTransport != null) {
+                connectedPrimaryTransport = preferredPrimaryTransport;
+                startService(SessionType.RPC, (byte) 0x00, false);
+            }else{
+                onTransportNotAccepted("No transports match requested primary transport");
+            }
+            // return;
+        }else if(requiresHighBandwidth){
```

Comment:
This means that a secondary transport will never be registered if this flag isn't set. This should not be the case, as the transport could have been started because of the logic in the `startService` function.

Starting a video service will time out with this logic if `requiresHighBandwidth` is set to false and Core does not allow this service on the primary transport.

Reply:
Changed logic to check for secondary transport listeners. A listener is added when the audio/video service have been determined to start on a transport but the app is either not registered on the transport or the transport has not been connected.